pkgs/by-name/ar/argc/package.nix

```diff
@@ -0,0 +1,64 @@
+{
+  lib,
+  rustPlatform,
+  stdenv, # Determin platforms.
+  fetchFromGitHub,
+  nixpkgsFun, # Re-evaluate Nixpkgs.
+  argc, # Get native argc for shell completion generation during cross-compilation.
+  installShellFiles,
+}:
+
+let
+  canExecuteHost = stdenv.buildPlatform.canExecute stdenv.hostPlatform;
+in
+rustPlatform.buildRustPackage rec {
+  pname = "argc";
+  version = "1.14.0";
+
+  src = fetchFromGitHub {
+    owner = "sigoden";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-Li/K5/SLG6JuoRJDz2DQoj1Oi9LQgZWHNvtZ1HVbj88=";
+  };
+
+  cargoHash = "sha256-D1T9FWTvwKtAYoqFlR2OmLRLGWhPJ9D8J7lq/QKcBoM=";
+
+  nativeBuildInputs = [ installShellFiles ] ++ lib.optional (!canExecuteHost) argc;
+
+  postInstall = ''
+    INSTALL_SHELL_FILE_PATH="${lib.optionalString canExecuteHost ''''${!outputBin}/bin''${PATH:+:}''}''${PATH-}"
+
+    PATH="$INSTALL_SHELL_FILE_PATH" installShellCompletion --cmd argc \
+      --bash <(argc --argc-completions bash) \
+      --fish <(argc --argc-completions fish) \
+      --zsh <(argc --argc-completions zsh)
+  '';
+
+  disallowedRequisites = lib.optional (!canExecuteHost) argc.__spliced.buildHost or argc;
```

Comment:
Why the or here? Either we don't need the value or the splicing is incorrect in which case things will not work either way.

Reply:
An alternative way is to add `argcNative` as an argument and specify it as `buildPackages.argc` inside `all-packages.nix`.

Reply:
Address by adding an argument `argcNative`,

Reply:
We can just use buildPackages.argc. That's pretty common to do and has less indirection.